src/web_info.rs

```diff
@@ -0,0 +1,43 @@
+use crate::error::AybError;
+use crate::templating::TemplateString;
+use serde::Deserialize;
+use url::Url;
+
+#[derive(Clone, Deserialize)]
+pub struct WebInfo {
+    base_url: Url,
+    endpoints: WebEndpoints,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct WebEndpoints {
+    profile: TemplateString,
+    confirmation: TemplateString,
+}
+
+impl WebInfo {
```

Comment:
I found this implementation to be quite elegant and compact! :)

Reply:
Thank youu! :)